.github/workflows/docker.yml

```diff
@@ -0,0 +1,75 @@
+        run: |
+          docker buildx build . --builder=container --cache-from=type=registry,ref=ghrc.io/datadog/catadog --load --platform linux/x86_64 -f ./Dockerfile --tag ghrc.io/datadog/catadog
+      - name: Test single-arch image (x86-64)
+        run: |
+          docker run --platform linux/x86_64 --rm ghrc.io/datadog/catadog /bin/sh -c 'true'
+          docker run --platform linux/x86_64 --rm ghrc.io/datadog/catadog ruby -e 'puts RUBY_DESCRIPTION'
+          docker run --platform linux/x86_64 --rm ghrc.io/datadog/catadog gem --version
+          docker run --platform linux/x86_64 --rm ghrc.io/datadog/catadog bundle --version
+          docker run --platform linux/x86_64 --rm -v "${PWD}":"${PWD}" -w "${PWD}" ghrc.io/datadog/catadog /bin/sh -c 'bundle install && bundle exec rake test'
+
+      # Build image for aarch64-linux, emulated under qemu
+      # 
+      # Tag image separately to avoid interference with caching and so that testing step can reference the image
+      - name: Enable aarch64 emulation (x86-64)
+        run: |
+          docker run --privileged --rm tonistiigi/binfmt --install arm64
+      - name: Build single-arch image (aarch64-linux)
+        run: |
+          docker buildx build . --builder=container --cache-from=type=registry,ref=ghrc.io/datadog/catadog --output=type=image,push=false --platform linux/aarch64 -f ./Dockerfile
+      - name: Tag single-arch image (aarch64-linux)
+        run: |
+          docker buildx build .  --builder=container --cache-from=type=registry,ref=ghrc.io/datadog/catadog --load --platform linux/aarch64 -f ./Dockerfile --tag ghrc.io/datadog/catadog
+      - name: Test single-arch image (aarch64-linux)
+        run: |
+          docker run --platform linux/aarch64 --rm ghrc.io/datadog/catadog /bin/sh -c 'true'
+          docker run --platform linux/aarch64 --rm ghrc.io/datadog/catadog ruby -e 'puts RUBY_DESCRIPTION'
+          docker run --platform linux/aarch64 --rm ghrc.io/datadog/catadog gem --version
+          docker run --platform linux/aarch64 --rm ghrc.io/datadog/catadog bundle --version
+          docker run --platform linux/aarch64 --rm -v "${PWD}":"${PWD}" -w "${PWD}" ghrc.io/datadog/catadog /bin/sh -c 'bundle install && bundle exec rake test'
+      
+      # Assemble multi-arch image for a combined push to the registry
+      # 
+      # Docker build is rerun, but build is fast because the layers are already cached
+      - name: Log in to the container registry
+        if: ${{ inputs.push }}
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghrc.io -u ${{ github.actor }} --password-stdin
+      - name: Build multi-arch image (x86-64, aarch64)
+        if: ${{ inputs.push }}
+        run: |
+          docker buildx build . --builder=container --cache-from=type=registry,ref=ghrc.io/datadog/catadog --output=type=image,push=true --build-arg BUILDKIT_INLINE_CACHE=1 --platform linux/x86_64,linux/aarch64 -f ./Dockerfile --tag ghrc.io/datadog/catadog
```

Comment:
Currently, the multi-arch image here that is being pushed to ghcr.io/Datadog/catadog GitHub Packages is not tagged with any gem / ruby version. See [this example](https://github.com/DataDog/images-rb/blob/main/.github/workflows/build-ruby.yml#L123) where the images are tagged with their corresponding ruby / jruby version. Since this workflow only runs for ruby 3.4, I did not see the point in adding the one version 3.4 tag; however, should there still be a tag for the pushed image? Perhaps corresponding to the `catadog` version?

Reply:
Based on Tony's recommendation, I added tags that correspond with the github.run_id of each image.